internal/dag/builder.go

```diff
@@ -556,7 +599,7 @@ func defaultBackendRule(be *v1beta1.IngressBackend) v1beta1.IngressRule {
 }
 
 // DAG returns a *DAG representing the current state of this builder.
-func (b *builder) DAG() *DAG {
+func (b *builder) DAG(secrets map[Meta]*v1.Secret, services map[Meta]*v1.Service) *DAG {
```

Comment:
Passing these two values into DAG so they can be passed back out as fields on DAG feels wrong. These fields are already avaialble to DAG's caller, builder.compute, so they can be returned directly from dag.compute without weaving them through DAG. I think this gives extra weight to the comments ealier about not keeping around the whole dag so we can access two additional fields added to DAG. Just keep the two fields directly, preferably wrapped in some object that has a ShouldUpdate method.